src/server/v2.0/handler/user.go

```diff
@@ -495,10 +495,18 @@ func validateUserProfile(user *commonmodels.User) error {
 		return errors.BadRequestError(nil).WithMessage("realname with illegal length")
 	}
 
-	if utils.IsContainIllegalChar(user.Realname, []string{",", "~", "#", "$", "%"}) {
+	if strings.ContainsAny(user.Realname, common.IllegalCharsInUsername) {
```

Comment:
here it checks characters in `Realname`, while the variable name is common.IllegalCharsIn**Username**. This might cause confusion.

Reply:
The username should be check, and the Realname is legacy code, just keep it as it was.